src/agent/firebase-controller.ts

```diff
@@ -224,26 +263,53 @@ export class FirebaseController implements Controller {
     this.bpRef = this.db.ref(`cdbg/breakpoints/${this.debuggeeId}/active`);
 
     let breakpoints = [] as stackdriver.Breakpoint[];
-    this.bpRef.on('child_added', (snapshot: firebase.database.DataSnapshot) => {
-      debuglog(`new breakpoint: ${snapshot.key}`);
-      const breakpoint = snapshot.val();
-      breakpoint.id = snapshot.key;
-      breakpoints.push(breakpoint);
-      callback(null, breakpoints);
-    });
-    this.bpRef.on('child_removed', snapshot => {
-      // remove the breakpoint.
-      const bpId = snapshot.key;
-      breakpoints = breakpoints.filter(bp => bp.id !== bpId);
-      debuglog(`breakpoint removed: ${bpId}`);
-      callback(null, breakpoints);
-    });
+    this.bpRef.on(
+      'child_added',
+      (snapshot: firebase.database.DataSnapshot) => {
+        debuglog(`new breakpoint: ${snapshot.key}`);
+        const breakpoint = snapshot.val();
+        breakpoint.id = snapshot.key;
+        breakpoints.push(breakpoint);
+        callback(null, breakpoints);
+      },
+      (e: Error) => {
+        debuglog(
+          'unable to listen to child_added events on ' +
+            `cdbg/breakpoints/${this.debuggeeId}/active. ` +
+            'Please check your database settings.'
+        );
+        callback(e, []);
+      }
+    );
+    this.bpRef.on(
+      'child_removed',
+      snapshot => {
+        // remove the breakpoint.
+        const bpId = snapshot.key;
+        breakpoints = breakpoints.filter(bp => bp.id !== bpId);
+        debuglog(`breakpoint removed: ${bpId}`);
+        callback(null, breakpoints);
+      },
+      (e: Error) => {
+        debuglog(
+          'unable to listen to child_removed events on ' +
+            `cdbg/breakpoints/${this.debuggeeId}/active. ` +
+            'Please check your database settings.'
+        );
+        callback(e, []);
+      }
+    );
   }
 
   stop(): void {
     if (this.bpRef) {
       this.bpRef.off();
       this.bpRef = undefined;
     }
+    try {
+      firebase.app('cdbg').delete();
```

Comment:
May be worth either storing the returned app from the initialize, or using a defined constant for the app name, 'cdbg' to ensure it's used the same in the initialize and this cleanup.

Reply:
Using a constant.